### PR TITLE
Adding requirements-local.txt support

### PR DESCRIPTION
**Pull request by craig-rueda**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
* Adds support for a local requirements.txt to be added by individuals in the event that they want to add extra Python modules for local testing

### REVIEWERS

@dpgaspar

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -114,9 +114,10 @@ ENTRYPOINT ["/usr/bin/docker-entrypoint.sh"]
 ######################################################################
 FROM lean AS dev
 
-COPY ./requirements-dev.txt ./docker/requirements-extra.txt /app/
+COPY ./requirements-dev.txt ./docker/requirements* /app/
 
 USER root
 RUN cd /app \
-    && pip install --no-cache -r requirements-dev.txt -r requirements-extra.txt
+    && pip install --no-cache -r requirements-dev.txt -r requirements-extra.txt \
```
Comment (nytai) [inline Dockerfile]: is there a reason to not just direct users to add extra requirements in `requirements-extra.txt`?

File: superset-frontend/package-lock.json [hunk 2]
```diff
@@ -10815,7 +10815,8 @@
         },
         "kind-of": {
           "version": "6.0.2",
-          "resolved": ""
+          "resolved": "https://registry.npmjs.org/kind-of/-/kind-of-6.0.2.tgz",
```
Comment (nytai) [inline superset-frontend/package-lock.json]: likely due to a newer npm version in docker. We should probably maintain an `.nvmrc` version file or equivalent that can be read from docker.

File: Dockerfile (hunk 1 above)
Comment (craig-rueda) [reply Dockerfile]: Yep, we don't want local users' changes being managed in Git

**Review by dpgaspar — APPROVED** [review]

**Comment by aspedrosa** [comment]
@craig-rueda For production I suppose that the docker target defined [here](https://github.com/apache/incubator-superset/blob/0.36/docker-compose.yml#L22) should be **lean** right?
However in this case we can't install database requirements such as **psycopg2**.
I'm thinking correctly?

**Comment by craig-rueda** [comment]
I think that for "production" it makes sense to build a custom image off of a "lean" image. We at Preset are currently building some infrastructure around this Dockerfile: https://hub.docker.com/repository/docker/preset/superset

For each master push, we currently build the lean target. In the near future, we will add support for the release branches as well, although this won't be possible to do completely retroactively as the Dockerfile is still relatively new.

**Pull request merged**